src/graphql-props.ts

```diff
@@ -0,0 +1,153 @@
+import { User } from "@src/entity/User";
+import { formatError } from "@src/error";
+import * as bcrypt from 'bcrypt';
+import { Context } from 'graphql-yoga/dist/types';
+import * as jwt from 'jsonwebtoken';
+import { getRepository, Repository } from 'typeorm';
+
+const typeDefs = `
+type Query {
+  hello: String
+  user(id: ID!): User!
+  users(count: Int, skip: Int): Users!
+}
+
+type Mutation {
+  login(email: String!, password: String!, rememberMe: Boolean): Login
+  createUser(user: UserInput!): User!
+}
+
+type User {
+  id: ID!
+  name: String!
+  email: String!
+  birthDate: String!
+  cpf: Int!
+}
+
+input UserInput {
+  name: String!
+  email: String!
+  password: String!
+  birthDate: String!
+  cpf: Int!
+}
+
+type Login {
+  user: User!
+  token: String!
+}
+
+type Users {
+  users: [User!]!
+  hasMore: Boolean!
+  skippedUsers: Int!
+  totalUsers: Int!
+}
+`;
+
+const getVerification = (context: Context) => {
+  const auth = context.request.get('Authorization')
+  if (!auth) {
+    throw formatError(401, 'You must be logged in', new jwt.JsonWebTokenError(''));
+  }
+
+  const token = auth.replace('Bearer ', '');
+
+  const verification = jwt.verify(token, process.env.JWT_SECRET);
+  return verification;
+};
+
+const resolvers = {
+  Query: {
+    hello: () => 'Hello!',
+    user: async (_, { id }, context: Context) => {
+      getVerification(context);
+
+      const userRepository: Repository<User> = getRepository(User);
+      let user: User;
+      try {
+        user = await userRepository.findOneOrFail({ id });
+
+      } catch (error) {
+        throw formatError(404, 'User not found');
+      }
+      const { password, ...userReturn } = { ...user };
+      return userReturn;
+    },
+    users: async (_, { count, skip }, context: Context) => {
+      getVerification(context);
+
+      count = count || 10;
+      skip = skip || 0;
+
+      const userRepository: Repository<User> = getRepository(User);
+      const [users, usersCount] = await Promise.all([
+        userRepository.find({ take: count, skip, order: { name: 'ASC' } }),
+        userRepository.count()
+      ]);
```

Comment:
Só mostrando como ficaria:
```suggestion
      const [users, usersCount] = await userRepository.findAndCount({ take: count, skip, order: { name: 'ASC' } });
```

Reply:
✅